pkl-gradle/src/main/java/org/pkl/gradle/spec/BasePklSpec.java

```diff
@@ -57,4 +57,8 @@ public interface BasePklSpec {
   Property<URI> getHttpProxy();
 
   ListProperty<String> getHttpNoProxy();
+
+  MapProperty<String, String> getExternalModuleReaders();
+
+  MapProperty<String, String> getExternalResourceReaders();
```

Comment:
I think we should make this `MapProperty<String, ExternalReader>`, where `ExternalReader` is defined as as record class in package `org.pkl.gradle`:

```java
public record ExternalReader(String executable, @Nullable List<String> arguments) implements Serializable {
  public ExternalReader(String executable) {
    this(executable, null);
  }
}
```

And I _believe_ we should add `@Nested` as an annotation here.

@netvl: is this the right Gradle thing to do?

Reply:
Ahh I was wondering how to do this. Had a real hard time searching the gradle docs to figure out how to do that, so if you have a doc handy I'd love a link.

Reply:
I updated this, but not really sure how to go about writing a test for this. Looking at the other pkl-gradle tests, it seems like they actually run evaluation, but in this case I'd need a working external reader executable to point at.